gating/update_dependencies/run

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash -xe
+
+# Update OSA SHA to head of stable/ocata
+
+osa_branch=stable/ocata
+
+# Get current head of stable/ocata
+git clone "https://github.com/openstack/openstack-ansible" ${WORKSPACE}/oa
+pushd ${WORKSPACE}/oa
+  git checkout $osa_branch
+  osa_sha="$(git log -n 1 --format=%H)"
+popd
+
+# Insert current SHA into functions.sh
+sed -i '/OSA_RELEASE:-/ c\
+export OSA_RELEASE=${OSA_RELEASE:-"'${osa_sha}'"}' scripts/functions.sh
+
+
+# Update rpc_release. The mainline branch should always be one version
+# ahead of the rc branch, as the next rc branch will be cut from mainline
+# post release.
+
+# This update script will only ever run against a mainine branch, as
+# dependencies should not change once an RC branch has been cut.
+
+# Get RC branch version
+rc_branch="${BRANCH}-rc"
+git fetch origin
+git show origin/${rc_branch} || {
+  echo "rc-branch ${rc_branch} not found, aborting."
+  exit 1
```

Comment:
Perhaps instead of exiting, we just assume that we do not need to edit the release version because there has never been a release? Otherwise the job will fail and the SHA update will never be done.

Reply:
Done